### ISSUE TITLE
Session/User augmentation not working

### ISSUE THREAD

**Issue by krinklesaurus** [nextauthjs/next-auth#7015]
### Question 💬

Hi,
I tried to follow https://next-auth.js.org/getting-started/typescript and https://next-auth.js.org/getting-started/example#extensibility but augmenting the user with custom properties is not working for me. It seems like my app is not even recognizing my augmented type since making breaking changes to `types/next-auth.d.ts` does not make any difference at all.

Here's what I have:

`types/next-auth.d.ts`

    import NextAuth, { DefaultSession } from "next-auth"

    declare module "next-auth" {
      /**
       * Returned by `useSession`, `getSession` and received as a prop on the `SessionProvider` React Context
       */
      interface Session {
        user: {
          /** The user's postal address. */
          id: string
          username: string
        } & DefaultSession["user"]
      }
    }

`./pages/api/auth/[...nextauth].js`

    import CredentialsProvider from "next-auth/providers/credentials"
    import NextAuth from "next-auth"

    export const authOptions = { 
      callbacks: {
        session({ session, token, user }) {
          return session // The return type will match the one returned in `useSession()`
        },
      },    
      // Configure one or more authentication providers
      providers: [
        CredentialsProvider({
            // The name to display on the sign in form (e.g. "Sign in with...")
            name: "Credentials",
            // `credentials` is used to generate a form on the sign in page.
            // You can specify which fields should be submitted, by adding keys to the `credentials` object.
            // e.g. domain, username, password, 2FA token, etc.
            // You can pass any HTML attribute to the <input> tag through the object.
            credentials: {
                username: { 
                    label: "Username", 
                    type: "text", 
                    placeholder: "" 
                },
                password: { 
                    label: "Password", 
                    type: "password" 
                }
            },        
            async authorize(credentials, req) {
                const user = {
                    id: "e86daa69-1851-44d9-aa17-6802b7e389c7", 
                    username: "blubb", 
                    password: "blubb",
                    name: "Lordius Blubb",
                    email: "blubb@blubb.io" 
                }

                console.log("returning "+JSON.stringify(user))              
                if (user) {
                    // Any object returned will be saved in `user` property of the JWT
                    return user
                } else {
                    // If you return null then an error will be displayed advising the user to check their details.
                    return null

                // You can also Reject this callback with an Error thus the user will be sent to the error page with the error message as a query parameter
                }
            }
        })    
        // ...add more providers here
      ],
    }

    export default NextAuth(authOptions)

`tsconfig.json`

    {
      "compilerOptions": {
        "target": "es5",
        "lib": ["dom", "dom.iterable", "esnext"],
        "allowJs": true,
        "skipLibCheck": true,
        "strict": true,
        "forceConsistentCasingInFileNames": true,
        "noEmit": true,
        "esModuleInterop": true,
        "module": "esnext",
        "moduleResolution": "node",
        "resolveJsonModule": true,
        "isolatedModules": true,
        "jsx": "preserve",
        "jsxImportSource": "@emotion/react",
        "incremental": true
      },
      "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "types/next-auth.d.ts"],
      "exclude": ["node_modules"]
    }

`./pages/_app.tsx`

    import * as React from 'react';

    import { CacheProvider, EmotionCache } from '@emotion/react';

    import { AppProps } from 'next/app';
    import Copyright from '../src/Copyright';
    import CssBaseline from '@mui/material/CssBaseline';
    import Head from 'next/head';
    import NavBar from '../src/NavBar';
    import { Session } from 'next-auth';
    import { SessionProvider } from "next-auth/react"
    import { ThemeProvider } from '@mui/material/styles';
    import createEmotionCache from '../src/createEmotionCache';
    import theme from '../src/theme';

    const clientSideEmotionCache = createEmotionCache();

    interface MyAppProps extends AppProps {
      emotionCache?: EmotionCache;
      session?: Session;
    }

    export default function MyApp(props : MyAppProps) {
      
      const pageProps = props;
      const emotionCache = clientSideEmotionCache
      const Component = props.Component
      const session = props.session

      return (
        <CacheProvider value={emotionCache}>
          <Head>
            <meta name="viewport" content="initial-scale=1, width=device-width" />
          </Head>
          <ThemeProvider theme={theme}>
            <CssBaseline />
            <SessionProvider session={session}>
            <NavBar />
            <Component {...pageProps} />
            <Copyright />
            </SessionProvider>
          </ThemeProvider>
        </CacheProvider>
      );
    }

Output of authorize is

        returning {"id":"e86daa69-1851-44d9-aa17-6802b7e389c7","username":"blubb","password":"blubb","name":"Lordius Blubb","email":"blubb@blubb.io"}

but output of 

    export async function getServerSideProps(context) {
        const session = await getServerSession(context.req, context.res, authOptions)
        console.log("getServerSideProps session: "+JSON.stringify(session))
      
        return { props: {} }
      }

is just

    getServerSideProps session: {"user":{"name":"Lordius Blubb","email":"blubb@blubb.io"},"expires":"2023-04-20T04:52:26.628Z"}

### How to reproduce ☕️

see above

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by arnekolja** [comment]
Same here. I've been trying to debug this for hours and hours …

**Comment by krinklesaurus** [comment]
here's a sample repo illustrating the issue: https://github.com/krinklesaurus/next-auth-test

@lluia @balazsorban44

**Comment by codinesh** [comment]
In facing same issue, type is not recognised by typescript in server components.

However new property appended to session is available in useSession on client components. But it's not available on getServerSession on server components

**Comment by balazsorban44** [comment]
This is unrelated to TypeScript. Check out the docs:

https://next-auth.js.org/getting-started/example#extensibility
https://next-auth.js.org/configuration/callbacks
https://next-auth.js.org/configuration/nextjs#getserversession

**Comment by arnekolja** [comment]
I agree with @krinklesaurus. I've read the docs and the issues over here for days, but whenever it's getting close to a solution, it just says RTFM. It's an endless loop.
